APMrover2/AP_MotorsUGV.cpp

```diff
@@ -591,6 +614,60 @@ void AP_MotorsUGV::output_skid_steering(bool armed, float steering, float thrott
     output_throttle(SRV_Channel::k_throttleRight, 100.0f * motor_right);
 }
 
+// output for custom configurations
+void AP_MotorsUGV::output_custom_config(bool armed, float steering, float throttle, float lateral)
+{
+    // exit immediately if the frame type is set to UNDEFINED
+    if (rover.get_frame_type() == FRAME_TYPE_UNDEFINED) {
+        return;
+    }
+
+    if (armed) {
+
+        // clear and set limits based on input
+        set_limits_from_input(armed, steering, throttle);
+
+        // constrain steering
+        steering = constrain_float(steering, -4500.0f, 4500.0f);
+
+        // reverse steering when throttle is negative
+        if (is_negative(throttle)) {
+            steering *= -1.0f;
+        }
+
+        // scale throttle, steering and lateral inputs to -1 to 1
+        const float scaled_throttle = throttle / 100.0f;
+        const float scaled_steering = steering / 4500.0f;
+        const float scaled_lateral = lateral / 100.0f;
+
+        float thr_str_ltr_out;
+        float thr_str_ltr_max = 1;
+        for (uint8_t i=0; i<AP_MOTORS_NUM_MOTORS_MAX; i++) {
+            thr_str_ltr_out = (scaled_throttle * _throttle_factor[i]) +
+                              (scaled_steering * _steering_factor[i]) +
+                              (scaled_lateral * _lateral_factor[i]);
+            if (fabsf(thr_str_ltr_out) > thr_str_ltr_max) {
+                thr_str_ltr_max = fabsf(thr_str_ltr_out);
+            }
+            float output_vectored = (thr_str_ltr_out / thr_str_ltr_max);
+
+            // send output for each motor
+            SRV_Channels::set_output_scaled(SRV_Channels::get_motor_function(i), 100.0f * output_vectored);
```

Comment:
This change to using set-output-scaled instead of output_throttle() means that users cannot use motors that rely on a "duty cycle" (for speed) and "relay" (for direction) input.  It would be better to stick with using output_throttle() if possible.  I think the relay library supports up to 4 relays so I think it can work.